### PR TITLE
RD-4048: Added component which is used to animate menu subitems

### PR DESCRIPTION
**Pull request by NorbertAmartus**
## Description

Changes consists of adding transition to the menu subitems, so that they won't "jump", while the sidebar is being expanded.

## Screenshots / Videos

Same before/after effect has been achieved as shown in [jira task](https://cloudifysource.atlassian.net/browse/RD-4048)

## Checklist

- [x] My code follows the UI Code Style.
- [ ] I verified that all tests and checks have passed.
- [x] I verified if that change requires a documentation update.
- [x] I added proper labels to this PR.

## Tests

 

N/A, as the changes are strictly styling related 🚀 

## Documentation

N/A

### CODE REVIEW

File: app/components/sidebar/SideBarItem.tsx [hunk 1]
```diff
@@ -36,28 +48,30 @@ const SideBarItem: FunctionComponent<SideBarItemProps> = ({
...
         height: '100%',
         width: expandedSidebarWidth,
         paddingTop: 13,
-        paddingLeft: subItem && 25,
+        paddingLeft: subItem && 15,
         ...style
     };
 
     return (
-        <SideBarItemWrapper className="sidebarItemWrapper">
```
Comment (NorbertAmartus) [inline app/components/sidebar/SideBarItem.tsx:48]: Just a note:
I've double checked and this `className` is not being used anywhere in the project

**Review by qooban — DISMISSED** [review]
Checked UI - looks very good! :-)

Styling is not my strong area, so I'd like to someone else review it. I'll nominate @kubama 🙂 

Please check bundle size limit as it seems exceeded.

**Review by kubama — COMMENTED** [review]
Looks very nice.

I only have one question - did you consider using https://react.semantic-ui.com/modules/transition/? I don't have much experience with this SUI component, but it looks like if it was targeted to solve similar kind of issues.

**Comment by NorbertAmartus** [comment]
> Looks very nice.
> 
> I only have one question - did you consider using https://react.semantic-ui.com/modules/transition/? I don't have much experience with this SUI component, but it looks like if it was targeted to solve similar kind of issues.

I did not, thanks for letting me know about this feature 🥳 
As I'm now looking at it, implementing that functionality would require us to restructure `Sidebar` component to propagate the `$expanded` prop across it's children

Because only with having that information we could determine which stage of the animation should be visible on the screen

Currently that issue is being bypassed by overriding components styles by the `Sidebar` component, depending on the inner `$expanded`  prop value

**Review by kubama — APPROVED** [review]

**Review by kubama — APPROVED** [review]

**Pull request merged**